cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -775,18 +854,25 @@ func (m *stateMachine) unlockDataRunKey() (stateFunc, error) {
 			return m.unlockDataFallbackKey, nil
 		}
 
+		// if we didn't even find the device to the point where it would have
+		// been identified as decrypted or unencrypted device, we could have
+		// just entirely lost ubuntu-data-enc, and we could still have an
+		// encrypted device, so instead try to unlock ubuntu-save with the
+		// fallback case, and then if that fails in a specific manner it will
+		// fallback again to locating the unencrypted ubuntu-save
+
 		// not an encrypted device, so nothing to fall back to try and unlock
 		// data, so just mark it as not found and continue on to try and mount
 		// an unencrypted ubuntu-save directly
-		return m.locateUnencryptedSave, nil
+		return m.unlockSaveFallbackKey, nil
```

Comment:
now we have two m.unlockSaveFallbackKey, nil  in this function, can we drop something ?

Reply:
this is done